File: src/main/java/net/gtaun/shoebill/constant/GameTextStyle.java
package net.gtaun.shoebill.constant;

// Created by marvin on 28.12.14 in project shoebill-api.
// Copyright (c) 2014 Marvin Haschker. All rights reserved.
public enum GameTextStyle {
    NORMALE_MIDDLE (0),
    NORMAL_BOTTOM_RIGHT (1),
    GHETTO_MIDDLE (2),
    THIN_MIDDLE (3),
    THIN_MIDDLE_TOP (4),
    WHITE_THIN_CENTER (5),
    THICK_TOP (6);

    public static GameTextStyle get(int value)
    {
        return values() [value];
    }

    private final int value;

    private GameTextStyle(int value)
    {
        this.value = value;
    }

    public int getValue()
    {
        return value;
    }
}

File: src/main/java/net/gtaun/shoebill/constant/GameTextStyle.kt
package net.gtaun.shoebill.constant

/**
 * This enum contains the possible styles for a GameText. This information has been taken from the offical SA:MP
 * Wikipedia page.
 *
 * @author MK124
 * @author Marvin Haschker
 */
enum class GameTextStyle private constructor(val value: Int) {
    NORMALE_MIDDLE(0),
    NORMAL_BOTTOM_RIGHT(1),
    GHETTO_MIDDLE(2),
    THIN_MIDDLE(3),
    THIN_MIDDLE_TOP(4),
    WHITE_THIN_CENTER(5),
    THICK_TOP(6);

    companion object : Collectable<GameTextStyle>, Findable<Int, GameTextStyle> {
        private val VALUES = values().map { it.value to it }.toMap()

        /**
         * Gets a [GameTextStyle] by its id.
         */
        @JvmStatic
        override operator fun get(value: Int): GameTextStyle? = VALUES[value]

        /**
         * Gets all available [GameTextStyle]s.
         */
        @JvmStatic
        override fun get(): Collection<GameTextStyle> = VALUES.values
    }
}
